lib/smart_proxy_acd/acd_runner.rb

```diff
@@ -0,0 +1,190 @@
+require 'smart_proxy_dynflow/runner/command_runner'
+require 'tempfile'
+require 'rest-client'
+require 'tmpdir'
+require 'socket'
+
+# rubocop:disable ClassLength
+
+module Proxy
+  module Acd
+    # Implements the AcdRunner to be used by foreman_remote_execution
+    class AcdRunner < Proxy::Dynflow::Runner::CommandRunner
+      DEFAULT_REFRESH_INTERVAL = 1
+
+      class << self
+        def parse_dynflow_settings(path)
+          return @dynflow_settings if defined? @dynflow_settings
+          if File.exist?(path)
+            @dynflow_settings = {}
+            YAML.load_file(path).each do |key, value|
+              @dynflow_settings[key.to_s] = value
+            end
+          end
+          @dynflow_settings
+        end
+
+        def parse_ssl_options
+          return @ssl_options if defined? @ssl_options
+
+          @ssl_options = {}
+          return @ssl_options unless URI.parse(@dynflow_settings['foreman_url']).scheme == 'https'
+
+          @ssl_options[:verify_ssl] = OpenSSL::SSL::VERIFY_PEER
+
+          private_key_file = @dynflow_settings['foreman_ssl_key'] || @dynflow_settings['ssl_private_key']
+          if private_key_file
+            private_key = File.read(private_key_file)
+            @ssl_options[:ssl_client_key] = OpenSSL::PKey::RSA.new(private_key)
+          end
+          certificate_file = @dynflow_settings['foreman_ssl_cert'] || @dynflow_settings['ssl_certificate']
+          if certificate_file
+            certificate = File.read(certificate_file)
+            @ssl_options[:ssl_client_cert] = OpenSSL::X509::Certificate.new(certificate)
+          end
+          ca_file = @dynflow_settings['foreman_ssl_ca'] || @dynflow_settings['ssl_ca_file']
+          @ssl_options[:ssl_ca_file] = ca_file if ca_file
+          @ssl_options
+        end
+      end
```

Comment:
Is this still needed? The Smart Proxy has options for this. Like `Proxy::SETTINGS.foreman_url`. Even better may be to use [ForemanRequest](https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/request.rb)

Reply:
thanks for this idea. we should handle this with another PR.